lib/common/socketpool.c

```diff
@@ -471,6 +471,9 @@ void h2o_socketpool_connect(h2o_socketpool_connect_request_t **_req, h2o_socketp
     size_t target = SIZE_MAX;
     h2o_linklist_t *sockets = NULL;
 
+    if (pool->timeout == 0 && !h2o_socketpool_is_global(pool))
```

Comment:
IIUC, the rationale behind skipping is that when `pool->timeout` is zero, no socket would have been added to the pool. Assuming that is the case:
* Do we need to call `h2o_socketpool_is_global` here? It is just a type of pool.
* Do we need to make changes to `h2o_socketpool_return` too? I think that function takes a lock, adds the socket to the pool, then destroys the ones that have expired.

Reply:
I've addressed the first point is addressed in 03bf672.

Regarding the second point, I notice that while it is true, it is unneeded by the proxy handler, because if `pool->timeout == 0` then `h2o_socketpool_can_keepalive` return false, indicating that the HTTP connection established to the backends will avoid doing keep alives.